src/v/config/configuration.cc

```diff
@@ -1123,6 +1123,48 @@ configuration::configuration()
       "Enable automatic partition rebalancing when new nodes are added",
       {.needs_restart = needs_restart::no, .visibility = visibility::user},
       false)
+
+  , partition_autobalancing_mode(
+      *this,
+      "partition_autobalancing_mode",
+      "Partition autobalancing mode",
+      {.needs_restart = needs_restart::no,
+       .example = "node_add_remove",
+       .visibility = visibility::user},
+      model::partition_autobalancing_mode::off,
+      {
+        model::partition_autobalancing_mode::off,
+        model::partition_autobalancing_mode::node_add_remove,
+        model::partition_autobalancing_mode::continuous,
+      })
+  , partition_autobalancing_node_availability_timeout_sec(
+      *this,
+      "partition_autobalancing_node_availability_timeout_sec",
+      "Node unavailability timeout that triggers moving partitions from the "
+      "node",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      15min)
+  , partition_autobalancing_max_disk_usage_percent(
+      *this,
+      "partition_autobalancing_max_disk_usage_percent",
+      "Disk usage threshold that triggers moving partitions from the node",
+      {.needs_restart = needs_restart::no, .visibility = visibility::user},
+      80,
+      {.min = 0, .max = 100})
+  , partition_autobalancing_tick_interval_ms(
+      *this,
+      "partition_autobalancing_tick_interval_ms",
+      "Partition autobalancer tick interval",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      30s)
+  , partition_autobalancing_movement_batch_size_bytes(
+      *this,
+      "partition_autobalancing_movement_batch_size_bytes",
+      "Total size of partitions that autobalancer is going to move in one "
+      "batch",
```

Comment:
Yes, that's true. The limit is across the cluster, but in practice movements will touch only a handful of nodes. I chose this value to be in line with the `raft_learner_recovery_rate` default value (100 MB/s per node) - so it will take around 50 seconds to complete the batch. We'll probably have to tweak these values after large-scale testing.

Reply:
I see, makes sense. It'd be interesting to extend the planner to be smart enough to make moves that abide by some per-node rules. It's probably fine for now since most clusters have a relatively low node count, but as we grow we'll probably want to revisit this.